src/client/common/tooltips/generateContent.js

```diff
@@ -3,23 +3,27 @@ const classNames = require('classnames');
 const _ = require('lodash');
 const config = require('./config');
 
-
 //Handle name related metadata fields
-const standardNameHandler = (pair) => makeTooltipItem(pair[1], 'Approved Name: ');
+const standardNameHandler = (pair) => makeTooltipItem(pair[1], 'Name: ');
 const standardNameHandlerTrim = (pair) => standardNameHandler(pair);
+const nameHandlerTrim = (pair, expansionFunction) => {
+  let revisedList = filterChemicalFormulas(pair[1]);
+  let shortArray = trimValue(revisedList, config.defaultEntryLimit);
+  let expansionLink = revisedList.length > config.defaultEntryLimit ?
+    h('div.more-link', { onclick: () => expansionFunction(pair[0]) }, 'more »') : h('div.error');
 
-const nameHandlerTrim = (pair) => {
-  let shortArray = filterChemicalFormulas(trimValue(pair[1], 3));
   return h('div.fake-paragraph', [
     h('div.field-name', 'Synonyms:'),
-    valueToHtml(shortArray, true)
+    valueToHtml(shortArray, true),
+    expansionLink
   ]);
 };
-const nameHandler = (pair) => {
+const nameHandler = (pair, expansionFunction) => {
```

Comment:
can you describe what expansionFunction is?

Reply:
expansionFunction is the function that is executed when the user clicks on the generated field. It triggers a series of functions which generate the expanded view of the field. 

In the above example, expansionFunction will allow the user to toggle between displaying 3 names and displaying all the names.